lib/sort_by_length.rb

```diff
@@ -1,7 +1,24 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  new_array = my_sentence.split
+  word_length = split_array.length
```

Comment:
You don't have a variable named `split_array` and I think you mean `new_array`.  Otherwise it's a nice BubbleSort.